src/engine/Distinct.cpp

```diff
@@ -36,17 +33,107 @@ VariableToColumnMap Distinct::computeVariableToColumnMap() const {
   return subtree_->getVariableColumns();
 }
 
+template <size_t WIDTH>
+cppcoro::generator<IdTable> Distinct::lazyDistinct(
+    cppcoro::generator<IdTable> originalGenerator,
+    std::vector<ColumnIndex> keepIndices,
+    std::optional<IdTable> aggregateTable) {
```

Comment:
Is there a good reason to not use the `bool yieldOnce` pattern from other operations?